test/widgets/action_sheet_test.dart

```diff
@@ -99,46 +107,244 @@ void main() {
     connection.prepare(httpStatus: 400, json: fakeResponseJson);
   }
 
-  group('AddThumbsUpButton', () {
-    Future<void> tapButton(WidgetTester tester) async {
-      await tester.ensureVisible(find.byIcon(ZulipIcons.smile, skipOffstage: false));
-      await tester.tap(find.byIcon(ZulipIcons.smile));
-      await tester.pump(); // [MenuItemButton.onPressed] called in a post-frame callback: flutter/flutter@e4a39fa2e
-    }
+  group('ReactionButtons', () {
+    final popularCandidates = EmojiStore.popularEmojiCandidates;
 
-    testWidgets('success', (tester) async {
-      final message = eg.streamMessage();
-      await setupToMessageActionSheet(tester, message: message, narrow: TopicNarrow.ofMessage(message));
+    group('popular emoji reactions;', () {
+      testWidgets('ensure all are shown', (tester) async {
+        final message = eg.streamMessage();
+        await setupToMessageActionSheet(tester, message: message, narrow: TopicNarrow.ofMessage(message));
 
-      connection.prepare(json: {});
-      await tapButton(tester);
-      await tester.pump(Duration.zero);
+        // Ensure all popular emoji buttons are shown.
+        final emojis = tester.widgetList<UnicodeEmojiWidget>(find.descendant(
+          of: find.descendant(
+            of: find.byType(ReactionButtons),
+            matching: find.byType(InkWell)),
+          matching: find.byType(UnicodeEmojiWidget)));
```

Comment:
This feels pretty deep into the implementation details of this UI. Given the comment above, let's just leave out this test case.